tool/gravity/cli/clusterupdate.go

```diff
@@ -404,6 +399,41 @@ func checkForUpdate(
 	return updateApp, nil
 }
 
+func getUpdatePackage(env *localenv.LocalEnvironment, installedPackage loc.Locator, updatePackagePattern string) (*loc.Locator, error) {
```

Comment:
I would add a bit of logs to this function, to log each scenario e.g. "Upgrading to version %v explicitly provided on CLI", "Upgrading to version %v from this cluster image tarball", "Upgrading to latest version of installed application %v".

Also, could you please write a unit test for this logic before merging? I'd probably move this somewhere to lib/update/utils and make it into a utility method that can be easily tested / write a unit test for it.